code/go/0chain.net/core/transaction/entity.go

```diff
@@ -153,7 +153,7 @@ func (t *Transaction) ExecuteSmartContract(address, methodName, input string, va
 }
 
 func (t *Transaction) Verify() error {
-	t.zcntxn.SetTransactionHash(t.Hash)
+	t.zcntxn.SetTransactionHash(t.Hash) //nolint:errcheck // checked in the next step
```

Comment:
[This errors](https://github.com/0chain/gosdk/blob/master/zcncore/transaction.go#L400) if the ransaction already exists. Returning this error should fail an attempt to send a transaction that already exists. This seems more useful to the user than a generic "transaction_verify_error" so return it.